### DISCUSSION TITLE
v1.6.0

### DISCUSSION THREAD

**Discussion by cpholguera** [OWASP/mastg#2408] [Announcements]
Following up on the [OWASP MASVS v2.0.0 Release](https://mas.owasp.org/MASVS/) we're excited to announce the release of the new OWASP MASTG version v1.6.0. This update includes a range of new features, including the first phase of the MASTG refactoring, MASVS color-coding, upgraded MAS Checklists (for OWASP MASVS v2.0.0 + MASTG v1.6.0), and much more. See below for a detailed list of changes.

We'd like to thank all of our loyal contributors and welcome our [new contributors](#new-contributors).

> Special thanks to [NowSecure](https://www.nowsecure.com) for their consistent high-impact contributions to the project, especially for the MASVS refactoring, the OWASP MASTG refactoring, the [OWASP MAS website](https://mas.owasp.org) and this MASTG v1.6.0 release and for continuing spreading the word about the OWASP MAS project.
> 
> 💙 Thanks to dvuln, eShard, OHRUS and devoteam Cyber Trust for their generous [donations](https://mas.owasp.org/donate/)!

<br>

_Carlos Holguera, Sven Schleier and Jeroen Beckers - OWASP MAS project_

<br>

NOTE: the OWASP MASTG v1.6.0 relies on the latest [MASVS v2.0.0](https://github.com/OWASP/owasp-masvs/releases/tag/v2.0.0)

Help us improve! [questions](https://github.com/OWASP/owasp-mastg/discussions/categories/q-a) | [ideas](https://github.com/OWASP/owasp-mastg/discussions/categories/ideas) | [contact](https://mas.owasp.org/contact/)

<br>

#### Introducing the MASVS v2 Colors

We're bringing official colors to the MASVS! The new colors will be used across the MASVS v2.0.0 and MASTG v2.0.0 to help users quickly identify the different control groups. We've also revamped certain areas of our website to make them more readable and easier to navigate as well as to prepare for what's coming with the MASTSG v2.0.0 (keyword: "atomic tests").

<img style="border-radius: 5px" alt="masvs_colors" src="https://user-images.githubusercontent.com/29175115/236796274-af239519-790a-48d3-9413-787c97eb887c.png">

#### MASVS

In the MASVS home page, the new colors will be used to highlight the different control groups.

<img style="border-radius: 5px" alt="masvs_home" src="https://user-images.githubusercontent.com/29175115/236796942-ef12bda7-9ee5-4f36-93f7-e2361a0a2b85.png">

The individual controls will also be color-coded to help users quickly identify the different control groups. We've also redesigned the control pages to make them more readable and easier to navigate.

<img style="border-radius: 5px"  alt="masvs_control" src="https://user-images.githubusercontent.com/29175115/236797051-9fc87c6c-67d6-40a1-b573-845661411bb4.png">

#### MASTG

Now, when you navigate to the MASTG tests, you'll see that they are categorized by platform (Android/iOS) as well as by MASVS category, also using our new colors in the sidebar. The colors will also be used to highlight the different control groups in the test description.

Each test now contains a header section indicating the platform, the MASVS v1.5.0 controls, and the MASVS v2.0.0 controls.

<img style="border-radius: 5px"  alt="mastg_test" src="https://user-images.githubusercontent.com/29175115/236797099-ac1bb81e-d5a5-474c-9d1d-a12e2d269381.png">

We've also introduced a new section called "Resources" which is automatically generated using the inline links within the MASTG pages and serve as a quick reference to the most important resources for each test.

**NOTE:** The MASTG tests themselves haven't changed yet, we're still working on the refactoring. For now we've simply split the tests into individual pages to make them easier to navigate and reference. This will facilitate the work on the refactoring and the introduction of the new atomic tests.

#### MAS Checklist

The MAS Checklist pages and the MAS checklist itself have also been updated to use the new colors to highlight the different control groups and to make them easier to navigate.

<img width="1515" alt="checklist_home" src="https://user-images.githubusercontent.com/29175115/236797227-69bc0297-504a-4aba-878b-a4744512f253.png">

When you click on a MASVS group you'll see a table listing the new MASVS v2.0.0 controls as well as the corresponding MASTG tests (v1.5.0) for both the Android and the iOS platforms.

<img style="border-radius: 5px"  alt="checklist_detail" src="https://user-images.githubusercontent.com/29175115/236797244-bd59a5db-eea6-4fbc-87be-6299e147bf3e.png">

**NOTE:** The checklist contains the old MASVS v1 verification levels (L1, L2 and R) which we are currently reworking into "security testing profiles". The levels were assigned according to the MASVS v1 ID that the test was previously covering and might differ in the upcoming version of the MASTG and MAS Checklist.

For the upcoming of the MASTG version we will progressively split the MASTG tests into smaller tests, the so-called "atomic tests" and assign the new MAS profiles accordingly.

<br>

We hope you like the new colors and the changes we've made to the website. We're looking forward to your feedback! Please use our GitHub Discussions to post any [questions](https://github.com/OWASP/owasp-mastg/discussions/categories/q-a) or [ideas](https://github.com/OWASP/owasp-mastg/discussions/categories/ideas) you might have. If you see something wrong please let us know by [opening a bug issue](https://github.com/OWASP/owasp-mastg/issues/new?assignees=&labels=%F0%9F%90%9B%20bug).

#### More News
* Website Redesign and Restructure by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2242
* Update Talks (Cybersec Chile) by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2275
* Add NSConnect 2022 Talk by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2302
* Add Guidelines to Contribute with Crackmes by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2303
* Added AppSec EU and US Talks  by @sushi2k in https://github.com/OWASP/owasp-mastg/pull/2385
* Update with MASVS v2 Release by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2397
* Added Case Study by NowSecure by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2402
* MASTG Transition Version by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2396
### 🧪 MASTG Test Cases
* Add static analysis details for Android keyboard cache by @DIvanov503 in https://github.com/OWASP/owasp-mastg/pull/2254
* Recommend Using conscrypt for Old Android API Levels by @rlatapy-luna in https://github.com/OWASP/owasp-mastg/pull/2340
* Deprecate Fragment Injection Test for MSTG-PLATFORM-2 by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2328
* Proofreading fixes 0x05d part 1 by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2351
* Proofreading fixes 0x05d part 2 by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2358
* Add Test for Android Pending Intents to 0x05h by @su-vikas in https://github.com/OWASP/owasp-mastg/pull/2300
* Add Test for Implicit Intent Injection (MSTG-PLATFORM-2) by @LukasMarckmiller in https://github.com/OWASP/owasp-mastg/pull/2056
* Add codesign/ldid to the test Determining Whether the App is Debuggable (MSTG-CODE-2) by @sohsatoh in https://github.com/OWASP/owasp-mastg/pull/2296
* Add otool command to 0x06i-Testing-Code-Quality-and-Build-Settings.md by @rsenet in https://github.com/OWASP/owasp-mastg/pull/2362
* [Phase 1] Refactor 0x05h-Testing-Platform-Interaction.md (@NowSecure) by @angrymuffinx in https://github.com/OWASP/owasp-mastg/pull/2286
* [Phase 1] Refactor 0x06j-Testing-Resiliency-Against-Reverse-Engineering.md by @iotaaxel in https://github.com/OWASP/owasp-mastg/pull/2321
* [Phase 1] Refactor 0x0**-Testing-Code-Quality.md by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2381
* [Phase 1] Refactor 0x06h-Testing-Platform-Interaction.md by @TheDauntless in https://github.com/OWASP/owasp-mastg/pull/2380
* [Phase 1] Refactor 0x0**-Testing-Resiliency-Against-Reverse-Engineering.md by @sushi2k in https://github.com/OWASP/owasp-mastg/pull/2382
* [Phase 1] Refactor 0x0**-Local-authentication.md by @TheDauntless in https://github.com/OWASP/owasp-mastg/pull/2377
* [Phase 1] Refactor 0x0**-Testing-Network-Communication.md by @sushi2k in https://github.com/OWASP/owasp-mastg/pull/2378
* [Phase 1] Refactor 0x0**-Testing-Cryptography.md by @sushi2k in https://github.com/OWASP/owasp-mastg/pull/2372
* [Phase 1] Refactor 0x0**-Testing-Data-Storage.md by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2379
### 📖 MASTG Testing Fundamentals
* Proofreading fixes 0x04b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2276
* Proofreading fixes 0x04c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2277
* Proofreading fixes 0x04f by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2279
* Proofreading fixes 0x04g by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2281
* Proofreading fixes 0x04e by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2278
* Proofreading fixes 0x04i by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2287
* Proofreading fixes part 1 0x05a by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2289
* Proofreading fixes part 2 0x05a by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2292
* Proofreading fixes part 1 0x05b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2293
* Proofreading fixes part 3 0x05b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2298
* Proofreading fixes part 3 0x05b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2299
* Proofreading fixes part 2 0x05b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2297
* Add iOS tcpdump instructions to 0x06b and related mitmproxy reference to 0x08a by @cgarst in https://github.com/OWASP/owasp-mastg/pull/2326
### ✨ MASTG Testing Techniques
* Proofreading fixes part 1 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2301
* Proofreading fixes part 2 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2304
* Proofreading fixes part 3 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2327
* Proofreading fixes part 4 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2330
* Proofreading fixes part 5 5x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2332
* Proofreading fixes part 6 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2333
* Proofreading fixes part 7 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2349
* Proofreading fixes part 8 0x05c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2350
### 🪄 MASTG Testing Tools
* Add Vulnerable App Example to 0x4h & 2 new Reference Apps by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2284
* Proofreading fixes 0x08b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2288
* Add steps for Frida snippet generation from JADX by @cgarst in https://github.com/OWASP/owasp-mastg/pull/2331
### ⚡ Automation
* Add Dynamic Pages Support by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2342
### 🎉 New Donators
* Add dvuln to God Mode Donators by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2343
* Add eShard to Good Samaritan Donators by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2347
* Add OHRUS to Good Samaritan Donators by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2361
* Add devoteam Cyber Trust to Honorable Benefactor Donators by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2405
### 🐞 Errata Corrections
* Fix typo by @CDuPlooy in https://github.com/OWASP/owasp-mastg/pull/2255
* Proofreading fixes 0x02b by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2272
* Proofreading fixes 0x02c by @Laancelot in https://github.com/OWASP/owasp-mastg/pull/2273
* Remove deprecated method from iOS sample code by @vtourraine in https://github.com/OWASP/owasp-mastg/pull/2325
* Fix Broken Link in 0x06c-Reverse-Engineering-and-Tampering.md by @rsenet in https://github.com/OWASP/owasp-mastg/pull/2363
### Other Changes
* Add tech writing training to style guide #2007 by @Amod02-prog in https://github.com/OWASP/owasp-mastg/pull/2251
* Add Paper Sicherheitsüberprüfung von mobilen iOS Apps nach OWASP (German) by @sushi2k in https://github.com/OWASP/owasp-mastg/pull/2256
* Update 1_How_Can_You_Contribute.md by @p2635 in https://github.com/OWASP/owasp-mastg/pull/2259
* Add Crackmes Tab by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2266
* Update nav bar by @p2635 in https://github.com/OWASP/owasp-mastg/pull/2264
* Update links for crackmes page by @TheDauntless in https://github.com/OWASP/owasp-mastg/pull/2274
* Fix download link of the Russian MAS Checklist by @x0000ff in https://github.com/OWASP/owasp-mastg/pull/2334
* Extend the status column in the MAS Checklists by @fujiokayu in https://github.com/OWASP/owasp-mastg/pull/2341
* Restrict GITHUB_TOKEN permissions by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2239
* Disable Restriction for GITHUB_TOKEN to be able to Release by @cpholguera in https://github.com/OWASP/owasp-mastg/pull/2407

## New Contributors
* @Amod02-prog made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2251
* @CDuPlooy made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2255
* @p2635 made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2259
* @DIvanov503 made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2254
* @Laancelot made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2272
* @iotaaxel made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2291
* @angrymuffinx made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2286
* @cgarst made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2331
* @vtourraine made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2325
* @x0000ff made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2334
* @rlatapy-luna made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2340
* @LukasMarckmiller made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2056
* @sohsatoh made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2296
* @rsenet made their first contribution in https://github.com/OWASP/owasp-mastg/pull/2362

**Full Changelog**: https://github.com/OWASP/owasp-mastg/compare/v1.5.0...v1.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/OWASP/owasp-mastg/releases/tag/v1.6.0'>v1.6.0</a>.</em>

**Comment by bl13pbl03p** [comment]
I would like to submit an adjusted version of the excel document where the "Pass / Fail / N/A" field has been repaired. Where can I submit the file? Cannot find the path within the repo to upload a file in a PR. 

Thank you in advance!

**Reply by cpholguera** [reply]
Hi @bl13pbl03p, the excel file is 100 % automatically generated. Here are the relevant scripts

https://github.com/OWASP/owasp-mastg/blob/master/tools/scripts/yaml_to_excel.py

https://github.com/OWASP/owasp-mastg/blob/master/tools/scripts/excel_styles_and_validation.py

I'd assume that the bug might be in the second script. If you could help us with it, that'd be awesome!

**Reply by bl13pbl03p** [reply]
Thanks for the quick reply @cpholguera! I will try 😎
